src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int SPLIT_LINE_STATUS = 0;
+    private static final int SPLIT_LINE_VALUE = 1;
+    private static final int SUPPLY = 0;
+    private static final int BUY = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] supplyAndBuy = {0, 0};
+        readFromFile(fromFileName, supplyAndBuy);
+        writerToFile(toFileName, supplyAndBuy);
```

Comment:
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeToFile(report, toFileName);
```

Reply:
If I have understood and implemented everything correctly, your implementation is better in terms of clean code, but it seems to me that this implementation will have memory problems with large amounts of data, since instead of reading information line by line, we read and store the entire block of information at once.